ambry-router/src/main/java/com.github.ambry.router/PutManager.java

```diff
@@ -151,14 +157,31 @@ public void onEvent(ByteBufferAsyncWritableChannel.EventType e) {
    */
   void submitPutBlobOperation(BlobProperties blobProperties, byte[] userMetaData, ReadableStreamChannel channel,
       FutureResult<String> futureResult, Callback<String> callback) {
+    String partitionClass = getPartitionClass(blobProperties);
     PutOperation putOperation =
         new PutOperation(routerConfig, routerMetrics, clusterMap, responseHandler, notificationSystem, userMetaData,
             channel, futureResult, callback, routerCallback, chunkArrivalListener, kms, cryptoService, cryptoJobHandler,
-            time, blobProperties, defaultPartitionClass);
+            time, blobProperties, partitionClass);
     putOperations.add(putOperation);
     putOperation.startReadingFromChannel();
   }
 
+  /**
+   * @param blobProperties the properties of the blob being put
+   * @return the partition class as required by the properties
+   */
+  private String getPartitionClass(BlobProperties blobProperties) {
+    String partitionClass = defaultPartitionClass;
```

Comment:
When implementing encryption support, Siva chose to not pass AccountService into the router and keep `isEncrypted` in the `BlobProperties`. Why did we choose to handle replication policy differently this time? Is it because we do not intend to persist the replication policy in the on-disk blob properties?

Reply:
I think with encrypted, `isEncrypted` _is_ a part of the properties (since the container may switch it on/off but the blob remains encrypted/unencrypted) so it may not be apples to apples. 

In this case, the replication policy is actually a property of the container and not the blob i.e. if the replication policy were to change, the partition that the blob is on will now conform to the new replication policy (and hence the blob).

Reply:
The replication policy is mostly a property of the partition num chosen, right? If my understanding is correct, changing the container's replicationPolicy would only affect new blobs uploaded to the container, like with encryption. However, I understand that you could potentially change the partition class of the partition a specific blob was uploaded to, so its also not really a "blob" level property like encryption

Reply:
Correct - not a blob level property